### ISSUE TITLE
Sorting issues latest git

### ISSUE THREAD

**Issue by agaida** [lxqt/qps#179]
##### Expected Behavior

As a matter of fact - the latest changes were big and __great__ - with a bit of a but.
Process sorting is much better now, but still a bit crappy sometimes - sorry, has to be blunt and straight. I want the visual output of qps reproducible and deterministic, right now it isn't

##### Current Behavior
Process sorting is not predictable right now fully - we should change it, the current state is not really a 2.0, but very close.

##### Possible Solution
fixing it

##### Steps to Reproduce (for bugs)
I have some problems to write up steps to reproduce reliable, because when changing the views filters/sorting are not fully set/reset/applied - so the only thing i can bring are several screenshots:

* first start in my working machine, does the testing on a laptop - not rigid enough. Still great, but not so easy to interpret  - tree view is marked, no sorting indicator visible - no discussions, might be the expected output

![](https://i.imgur.com/FEDspPe.png)

the output i would expect first hand:
![](https://i.imgur.com/DO6OYvP.png)

Reason - it would look a bit irritating otherwise:
![](https://i.imgur.com/Nj1442A.png)

(like this)
![](https://i.imgur.com/qHJR9JN.png)

As i wrote, great work, but a little fine tuning is still needed. To be honest  - we can release as 2.0 without a problem, but if it would be easy to solve as a fast shot i would prefer it :D

**Comment by tsujan** [comment]
> the output i would expect first had:

See https://github.com/lxqt/qps/pull/177 → (4). Another PR is needed to make an unsorted table impossible -- I'd considered it.

I didn't understand the rest (please be more specific if the following explanation isn't enough) but if you mean sorting other columns when the tree is visible, it has this logic:

First the roots of the tree are considered but if they have the same value, the root PID is considered.; then the first branches are considered but if they have the same value...; then the second branches but if...; and so on for deeper branches.

In your last screenshot, 9 and 1 belong to the same branch depth, so, 9 is above 1. But "-", above 9, belongs to the root "1", so it has priority and should be above 9. In short, the screenshot shows a correct sorting.

Use MEM, USER or NICE and you'll see the logic.

I haven't seen any sorting issue.

**Comment by tsujan** [comment]
> we can release as 2.0

Please not until https://github.com/lxqt/qps/pull/177 and a PR after it (for making an unsorted table impossible) are merged!

**Comment by tsujan** [comment]
And please don't be surprised if you see, when the sort indicator is downward, NICE is sorted from low to high but MEM and %CPU from high to low. KSysGuard and gnome-system-monitor do the same. Perhaps there is a hidden consensus about that.

**Comment by agaida** [comment]
my main objection was the unsorted thing - with this impossile i would call qps nearly perfect :)

**Comment by tsujan** [comment]
> my main objection was the unsorted thing

I'm working on it; it shouldn't be hard to implement.

> with this impossile i would call qps nearly perfect :)

Although its code is still a mess, I think the app is a nice process manager now. I can recommend it outside LXQt too.

The code cleanup might take months, *at least*.

**Comment by agaida** [comment]
Hey, the base for qps was layed down and written in the last century iirc - so what do you expect :D

**Comment by tsujan** [comment]
> Hey, the base for qps was layed down and written in the last century

The original author did a great job; there's no doubt in that. But his code style was a mess for all centuries ;) His GUI codes weren't good either.

BTW, I fixed the unsorted table right now; hopefully I've made no mistake.
